Source/Engine/Core/Memory/BumpFastAllocation.h

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2012-2024 Wojciech Figat. All rights reserved.
+
+#pragma once
+
+#include "Allocation.h"
+#include "Engine/Core/Log.h"
+#include "Engine/Core/Collections/BitArray.h"
+
+ // This flag enables safety checks for freed memory.
+#define ASSERT_SAFE_BUMP_ALLOC 0
+
+#if ASSERT_SAFE_BUMP_ALLOC
+#include "Engine/Core/Collections/HashSet.h"
+#endif
+
+/// <summary>
+/// Memory allocation policy with the fastest possible allocation and deallocation.
+/// It uses a simple bump allocator that allocates memory in a linear fashion with alignment support.
+/// It does not synchronize access to the memory block, so it is not thread-safe.
+/// Context must be reset manually between uses.
+/// </summary>
+class BumpFastAllocation
```

Comment:
I think `BumpAllocation` name would make more sense.

Reply:
I wanted to have 2 bump allocators: with and without safe threading. Thus two of them: `BumpFactAlloc` and `BumpSafeAlloc`.